music/src/core/soundfont.ts

```diff
@@ -0,0 +1,377 @@
+/**
+ * Module for loading and playing SoundFont instrument samples.
+ *
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports
+ */
+import * as Tone from 'tone';
+import * as constants from './constants';
+
+/**
+ * Instrument sample pitch and velocity.
+ */
+export interface SampleInfo {
+  pitch: number;
+  velocity: number;
+}
+
+/**
+ * Specification for a sampled instrument. Samples must exist for all pitches
+ * between `minPitch` and `maxPitch` at all velocities in `velocities` (unless
+ * `velocities` is undefined, in which case only a single sample exists for each
+ * pitch. Each sample consists of a note sustained for `durationSeconds` then
+ * released, ending after `releaseSeconds` additional seconds.
+ *
+ * @param name Name of the instrument.
+ * @param minPitch The minimum MIDI pitch sampled.
+ * @param maxPitch The maximum MIDI pitch sampled.
+ * @param durationSeconds Length of each sample in seconds, not including the
+ * release.
+ * @param releaseSeconds Length of the release for each sample in seconds.
+ * @param percussive If true, the sample is considered percussive and will
+ * always be played in its entirety.
+ * @param velocities (Optional) The set of velocities sampled.
+ */
+export interface InstrumentSpec {
+  name: string;
+  minPitch: number;
+  maxPitch: number;
+  durationSeconds: number;
+  releaseSeconds: number;
+  percussive: boolean;
+  velocities?: number[];
+}
+
+/**
+ * Sampled instrument. Must be initialized and samples must be pre-loaded using
+ * the `loadSamples` method before any notes can be played.
+ */
+export class Instrument {
+  private FADE_SECONDS = 0.1;
+
+  private readonly baseURL: string;
+  // tslint:disable-next-line:no-any
```

Comment:
// tslint:disable-line on same line preferred here and elsewhere

Reply:
Done.